doc/developer_guild.md

```diff
@@ -0,0 +1,471 @@
+# Extend Functionalities of GoTorch
+
+At the very core of PyTorch is *libtorch*, which is a C++ library that contains
+three levels of abstraction:
+
+1. *Native function*s are low-level functions, most of which operate tensors.
+   They are the basic mathematical build blocks of deep neural networks.
+1. *Functional*s are high-level functions built on native functions, each
+   functional may call several native functions.
+1. *Module*s are c++ classes/structs/templates that encapsulate multiple native
+   functions or functionals.
+
+Modules and functionals form the Pytorch C++ frontend.  AI practitioners can
+directly use libtorch modules or functionals to build their AI applications.
+
+libtorch contains about 1600 native functions, they are the foundations of
```

Comment:
This paragraph highly duplicate with the `wrap_native_functions.md`. I would recommend to remove it.

Reply:
Done.